lib/archethic/self_repair/notifier.ex

```diff
@@ -1,104 +1,172 @@
 defmodule Archethic.SelfRepair.Notifier do
-  @moduledoc false
+  @moduledoc """
+  Process to handle repair in case of topology change by trying to replicate transactions to new shard composition.
+
+  When a node receive a topology change due to the unavailability of a node,
+  we compute the new election for the already stored transactions.
+
+  Hence, a new shard might me formed as we notify the new transactions to the
+  new storage nodes
+
+  ```mermaid
+  flowchart TD
+      A[Node 4] --x|Topology change notification| B[Node1]
+      B --> | List transactions| B
+      B -->|Elect new nodes| H[Transaction replication]
+      H -->|Replicate Transaction| C[Node2]
+      H -->|Replicate Transaction| D[Node3]
+  ```
+
+  """
 
   use GenServer
 
-  #  alias Archethic.Crypto
-  #
-  #  alias Archethic.Election
-  #
+  alias Archethic.Crypto
+
+  alias Archethic.Election
+
   alias Archethic.PubSub
 
-  #  alias Archethic.P2P
-  #  alias Archethic.P2P.Message.ReplicateTransactionChain
+  alias Archethic.P2P
+  alias Archethic.P2P.Message.ReplicateTransaction
   alias Archethic.P2P.Node
 
-  #  alias Archethic.TaskSupervisor
-  #
-  #  alias Archethic.TransactionChain
-  #  alias Archethic.TransactionChain.Transaction
-  #
-  #  alias Archethic.Utils
+  alias Archethic.TaskSupervisor
+
+  alias Archethic.TransactionChain
+  alias Archethic.TransactionChain.Transaction
+  alias Archethic.TransactionChain.Transaction.ValidationStamp
+
+  alias Archethic.Utils
+
+  require Logger
 
   def start_link(args \\ []) do
     GenServer.start_link(__MODULE__, args)
   end
 
   def init(_) do
     PubSub.register_to_node_update()
-    {:ok, []}
+    {:ok, %{notified: %{}}}
```

Comment:
if there is process crash??, and in a situation where all nodes  waiting for each other for node_update:  message?

Reply:
This state is just to avoid multiple node_update message for a given node, to avoid multiple repair in a close time.

Reply:
If the process crashes, it will restart with an empty state, and wait another message of unavailability of the node